oauth2client/contrib/flask_util.py

```diff
@@ -443,7 +443,12 @@ def credentials(self):
 
     def has_credentials(self):
         """Returns True if there are valid credentials for the current user."""
-        return self.credentials and not self.credentials.invalid
```

Comment:
I suspect it's basically impossible.

In case you needed more understanding of why I try so hard to minimize the public interface of a library released into the wild...